recipe/meta.yaml

```diff
@@ -10,17 +10,23 @@ source:
   fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: {{ sha256sum }}
+  patches:
+    - patches/0001-remove-setup_requires.patch
 
 build:
-  number: 1000
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
   skip: True  # [win32 or linux32]
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - cython
+    - wheel
   host:
+    - cython
```

Comment:
Is cython needed both on build and host? Also, shouldn't wheel be just on host?

Reply:
cymem specifies it as a [requirement](https://github.com/explosion/cymem/blob/master/requirements.txt) (though with outdated guard against then-unknown updates), but does not enforce this when pip-installing or when importing the package.

But since cymem [defines](https://github.com/explosion/cymem#cymem-a-cython-memory-helper) itself as "cymem provides two small memory-management helpers for Cython", I think it's fair to say that this dependency is necessary on host. I can test if it builds without...?

Wheel is specifically necessary at install-time, also with outdated version guards (which is why i had to write the patch in the first place). It's a bit redundant to specify it at all because installing python brings in wheel as a dependency, but I thought "explicit is better than implicit", and cymem [added it](https://github.com/explosion/cymem/commit/fa8e5bb9104bd8a27918103f0e5b39fb3f12bc6f) explicitly too...

Reply:
I agree that both wheel and cython are needed but they should be in `host` not `build`.  If they appear in both there will be two Python interpreters and `site-packages` directories, one in the build environment and one in the host.  This is asking for confusion, it is much better to include this only in `host`

Reply:
Clearly, I have much to learn, haha...
I thought build and host are completely independent for this. Where would the two interpreters/site-packages live?

Reply:
Yes, build and host can be confusing and is an advanced topic.  During a build conda-build creates two separate conda environments, one containing the requirements in `build` and the second containing those in `host`.  On linux the `build` environment is stored in `...conda-bld/pkg_name_timestamp/_build_env` where as the host environment is in `.../conda-build/pkg_name_timestamp/_h_env_place..../` although the locations are implementation details not formal specifications.  You can see these two environments being created during the build by looking for `## Package Plan ##` The two environments are stacked so that binaries in the `host` environment appear before those in the `build` environment.  

After running the build script, the `host` environment is checked for new files and any new files go into the package.  The `build` environment is not checked for new files, if something is added there that is too bad.

Reply:
Thanks for the explanation! :)

Anything left to do here?

Reply:
LGTM